reactor-pool-micrometer/src/main/java/reactor/pool/introspection/micrometer/MicrometerMetricsRecorder.java

```diff
@@ -0,0 +1,120 @@
+
+import java.util.concurrent.TimeUnit;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.DistributionSummary;
+import io.micrometer.core.instrument.MeterRegistry;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.Timer;
+
+import reactor.pool.PoolMetricsRecorder;
+
+final class MicrometerMetricsRecorder implements PoolMetricsRecorder {
+
+	private final String        metricsPrefix;
+	private final MeterRegistry meterRegistry;
+
+	private final Timer               allocationFailureTimer;
+	private final Timer               allocationSuccessTimer;
+	private final DistributionSummary destroyedMeter;
+	private final Counter             recycledCounter;
+	private final Counter             recycledNotableFastPathCounter;
+	private final Counter             recycledNotableSlowPathCounter;
+	private final DistributionSummary resetMeter;
+	private final DistributionSummary resourceSummaryIdleness;
+	private final DistributionSummary resourceSummaryLifetime;
+
+	MicrometerMetricsRecorder(String metricsPrefix, MeterRegistry registry) {
+		this.metricsPrefix = metricsPrefix;
+		this.meterRegistry = registry;
+
+		allocationSuccessTimer = this.meterRegistry.timer(
+			DocumentedPoolMeters.ALLOCATION.getName(this.metricsPrefix),
+			Tags.of(DocumentedPoolMeters.AllocationTags.OUTCOME_SUCCESS));
+		allocationFailureTimer = this.meterRegistry.timer(
+			DocumentedPoolMeters.ALLOCATION.getName(this.metricsPrefix),
+			Tags.of(DocumentedPoolMeters.AllocationTags.OUTCOME_FAILURE));
+
+		resetMeter = this.meterRegistry.summary(
+			DocumentedPoolMeters.RESET.getName(this.metricsPrefix));
+		destroyedMeter = this.meterRegistry.summary(
+			DocumentedPoolMeters.DESTROYED.getName(this.metricsPrefix));
+
+		recycledCounter = this.meterRegistry.counter(
+			DocumentedPoolMeters.RECYCLED.getName(this.metricsPrefix));
+
+		recycledNotableFastPathCounter = this.meterRegistry.counter(
+			DocumentedPoolMeters.RECYCLED_NOTABLE.getName(this.metricsPrefix),
+			Tags.of(DocumentedPoolMeters.RecycledNotableTags.PATH_FAST));
+		recycledNotableSlowPathCounter = this.meterRegistry.counter(
+			DocumentedPoolMeters.RECYCLED_NOTABLE.getName(this.metricsPrefix),
+			Tags.of(DocumentedPoolMeters.RecycledNotableTags.PATH_SLOW));
+
+		resourceSummaryLifetime = this.meterRegistry.summary(
+			DocumentedPoolMeters.SUMMARY_LIFETIME.getName(this.metricsPrefix));
+		resourceSummaryIdleness = this.meterRegistry.summary(
+			DocumentedPoolMeters.SUMMARY_IDLENESS.getName(this.metricsPrefix));
+	}
+
+	@Override
+	public void recordAllocationSuccessAndLatency(long latencyMs) {
+		allocationSuccessTimer.record(latencyMs, TimeUnit.MILLISECONDS);
+	}
+
+	@Override
+	public void recordAllocationFailureAndLatency(long latencyMs) {
+		allocationFailureTimer.record(latencyMs, TimeUnit.MILLISECONDS);
+	}
+
+	@Override
+	public void recordResetLatency(long latencyMs) {
+		resetMeter.record(latencyMs);
```

Comment:
Should not this and destroy be a `Timer` instead of a `DistributionSummary`?
A `Timer` is basically a `DistributionSummary` except it records elapsed time, it has similar features but it also has the ability to scale the time to the unit of the choice of the registry (i.e.: ms vs. s).

Reply:
interesting, in that case yes I'll convert these to `Timer`s